algorithms/linfa-ftrl/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+//! # Follow the regularized leader - proximal
+//!
+//! ## The Big Picture
+//!
+//! `linfa-ftrl` is a crate in the [`linfa`](https://crates.io/crates/linfa) ecosystem, an effort to create a toolkit for classical Machine Learning implemented in pure Rust, akin to Python's `scikit-learn`.
+//!
+//! ## Current state
+//! `linfa-ftrl` provides a pure Rust implementation of an [algorithm](struct.FollowTheRegularizedLeader.html).
+//!
+//! ## Examples
+//!
+//! There is an usage example in the `examples/` directory. To run, use:
+//!
+//! ```bash
+//! $ cargo run --example winequality
+//! ```
+//!
+mod algorithm;
+mod error;
+mod hyperparams;
+
+use crate::hyperparams::FtrlParams;
+pub use algorithm::Result;
+pub use error::FtrlError;
+use linfa::Float;
+use ndarray::Array1;
+use ndarray_rand::RandomExt;
+use rand::distributions::Uniform;
+
+#[cfg(feature = "serde")]
+use serde_crate::{Deserialize, Serialize};
+
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+pub struct FollowTheRegularizedLeader<F: Float> {
```

Comment:
Rename this to `FTRL`, since the full name is a bit long. Also add docs and an example to this type.